core/common/assets/js/api/core/data.js

```diff
@@ -0,0 +1,442 @@
+		// eg: 'documents/:documentId/elements/:elementId' and args { documentId: 4123 }.
+		// result: 'documents/4123/elements'
+		if ( endpoint.includes( '/:' ) ) {
+			endpoint = endpoint.substring( 0, endpoint.indexOf( '/:' ) );
+		}
+
+		return endpoint;
+	}
+
+	/**
+	 * Function endpointToCommand().
+	 *
+	 * Convert endpoint to command.
+	 *
+	 * TODO: Find a better solution
+	 *
+	 * @param {string} endpoint
+	 * @param {object} args
+	 *
+	 * @return {string} command
+	 */
+	endpointToCommand( endpoint, args = {} ) {
+		let command = endpoint,
+			commandFound = !! $e.data.commands[ endpoint ];
+
+		const endpointParts = endpoint.split( '/' );
+
+		// Assuming the command maybe index.
+		if ( ! commandFound && $e.data.commands[ endpoint + '/index' ] ) {
+			command = endpoint + '/index';
+			commandFound = true;
+		}
+
+		// Maybe the it has 'id' as last endpointPart.
+		if ( ! commandFound ) {
+			// Remove last parameter.
+			const lastParameter = endpointParts[ endpointParts.length - 1 ],
+				assumedCommand = command.replace( '/' + lastParameter, '' );
+
+			// If assumed command ( without last parameter ) exist.
+			if ( $e.data.commands[ assumedCommand ] ) {
+				command = assumedCommand;
+				commandFound = true;
+
+				if ( ! args.query ) {
+					args.query = {};
+				}
+
+				// Warp with 'id'.
+				args.query.id = lastParameter;
+			}
+		}
+
+		return command;
+	}
+
+	/**
+	 * Function prepareHeaders().
+	 *
+	 * @param {DataTypes} type
+	 * @param {RequestData} requestData
+	 *
+	 * @return {{}} params
+	 */
+	prepareHeaders( type, requestData ) {
+		/* global wpApiSettings */
+		const nonce = wpApiSettings.nonce,
+			params = {
+				credentials: 'include', // cookies is required for wp reset.
+			},
+			headers = { 'X-WP-Nonce': nonce };
+
+		/**
+		 * Translate:
+		 * 'create, delete, get, update' to HTTP Methods:
+		 * 'GET, POST, PUT, PATCH, DELETE'
+		 */
+		const allowedMethods = this.getAllowedMethods( type ),
+			method = this.getHTTPMethod( type );
+
+		if ( 'GET' === method ) {
+			Object.assign( params, { headers } );
+		} else if ( allowedMethods ) {
+			Object.assign( headers, { 'Content-Type': 'application/json' } );
+			Object.assign( params, {
+				method,
+				headers,
+				body: JSON.stringify( requestData ),
+			} );
+		} else {
+			throw Error( `Invalid type: '${ type }'` );
+		}
+
+		return params;
+	}
+
+	/**
+	 * Function fetch().
+	 *
+	 * @param {DataTypes} type
+	 * @param {RequestData} requestData
+	 *
+	 * @return {{}} params
+	 */
+	fetch( type, requestData ) {
+		requestData.cache = 'miss';
+
+		const params = this.prepareHeaders( type, requestData ),
+			useCache = 'get' === type && ! requestData.args.options?.refresh;
+
+		if ( useCache ) {
+			const cachePromise = this.cache.getAsync( requestData );
+
+			if ( cachePromise ) {
+				return cachePromise;
+			}
+		}
+
+		return new Promise( async ( resolve, reject ) => {
```

Comment:
Why does it need to be async?

Reply:
because this is `fetch` function, and original fetch return promise by default, this is just wrapper.